rekolekcje-engine/src/main/java/pl/oaza/warszawa/dor/rekolekcje/api/participants/domain/ParticipantCreator.java

```diff
@@ -95,6 +106,31 @@ private Experience fromValue(ExperienceValue experienceValue) {
         .stepsPlannedThisYear(experienceValue.getStepsPlannedThisYear())
         .celebrationsTaken(experienceValue.getCelebrationsTaken())
         .celebrationsPlannedThisYear(experienceValue.getCelebrationsPlannedThisYear())
+        .historicalRetreats(extractHistoricalRetreats(experienceValue))
+        .build();
+  }
+
+  private Set<RetreatTurn> extractHistoricalRetreats(ExperienceValue experienceValue) {
+    if (experienceValue.getHistoricalRetreats() == null) {
+      return null;
```

Comment:
W praktyce powinno zwracać się nulla, czy pusty Set?

Reply:
Różnie. W praktyce powinno się zwracać nulla, kiedy jest null. Wciąż się zastanawiam nad tymi nullami w projekcie, niedługo prawdopodobnie wszystko przerobię na `Optional`e. Z tymi nullami to jest tak, że w bazie jest null, bo nie może być np 0, null pokazuje nam, że wartość nie została wypełniona, a nie że została wypełniona zerem. Poza tym w przypadku DTO wartość 0 i null mówi coś innego. W przypadku kolekcji to się zastanawiam. Zdecydowanie wolałbym pusty Set, ale wydaje mi się, że jak przychodzi request z backendu, to to pole serializuje się do nulla, a nie do pustego seta. Może powinniśmy to jakoś wymuszać, żeby jednak serializować do pustej kolekcji, albo żeby frontend zawsze tworzył kolekcję, ale wtedy albo komplikujemy serializację/deserializację, albo dodajemy dodatkową logikę po stronie frontendu. Ale z drugiej strony mamy dużo null-checków po stronie API, więc też niezbyt dobrze. Kolejny z wielu tematów do przemyślenia. Dziś już trochę o tym czytałem i rozmawiałem z architektem w pracy, mam jeszcze w planach w najbliższych dniach rozdział w effective java, może coś wymyślę ;) Popróbuję innych pomysłów trochę, jak skończę tego mojego ficzera, i zobaczymy ;)